### ISSUE TITLE
Make street+destination box expandable

### ISSUE THREAD

**Issue by polarbearing** [osmandapp/Osmand#3776]
As already discussed in #3767, I propose to have the box for street name and destination expandable. 
I like how both street and destination are currently combined in this box, separated with the >>, however roads can have multiple destinations signposted, which do not fit one line. 
Often, the destinations are sorted from far to near, so the nearest destination, e.g. the next exit, is hidden.
The expansion should be optional in pull-down fashion, and fit the size of the text.
The user control could either be a triangle in the right corner of the box itself to toggle the behaviour, or an option in the vehicle-specific profile. 
As I expect that most useful on a motorway, a direct access to the control would be preferred. 
Current situation and mockup below. 
![screenshot_20170316_more_destination](https://cloud.githubusercontent.com/assets/8266355/26032360/ef4358bc-3891-11e7-9842-83d0b0db5103.png)
![screenshot_20170316_mockup_destination](https://cloud.githubusercontent.com/assets/8266355/26032363/f53b130e-3891-11e7-9821-5d9fe6580f34.png)

**Comment by bitboy85** [comment]
Another suggestions: Make the box a two liner. first for the road which is always visible, the second one for Destination which is shown if a Destination is tagged.
I know that this still might hide some Information, but the Driver doesn't need to interact and from my personal experience, the first 2 or 3 destinations make it clear enough where to go. As an alternative, it could be auto-expandable.

**Comment by polarbearing** [comment]
> Make the box a two liner.

No it should only take screen space when needed. 

> the first 2 or 3 destinations make it clear enough where to go

no, if you are looking for an exit which is typically at the end of the list, you are missing this. The first 2-3 destinations are the distant target, the last one is the nearest. 

> auto-expandable

yes.

**Comment by bitboy85** [comment]
> No it should only take screen space when needed.

Thats what i said, "which is shown if a Destination is tagged" if there is no, the second line should be hidden.

> no, if you are looking for an exit which is typically at the end of the list, you are missing this. The first 2-3 destinations are the distant target, the last one is the nearest.

I had in mind that osmand is choosing the exit for you, so you just have to take the right one and usually the first 2 - 3 Destination is enough to identify the Exit provided by osmand.

auto-expandable might be an issue if there exists signs with many many destinations (or other informations like Airports and Special places) on it so the text box covers most of the screen.

**Comment by polarbearing** [comment]
> text box covers most of the screen

yes that's why it is necessary to have a toggle to switch it on/off. Naturally the number of destinations is limited to what fits on the related traffic sign. It is not meant to show flight schedules. 

> first 2 - 3 Destination is enough to identify the Exit provided by osmand

It should work both in navigation mode and even more importantly in non-navigation mode. In particular to monitor which exits you are passing, the sequence of different near-targets is important, not so much the constant far-target (however that is relevant to check that you are driving on the correct road)
